### PR TITLE
Astral travel

### PR DESCRIPTION
**Pull request by jandersson**
Working implementation of astral travel. Starts at any Grazhir shard and travels to another using fixed harness amounts

### CODE REVIEW

File: bescort.lic [hunk 1]
```diff
@@ -524,6 +530,101 @@ class Bescort
...
+    if Flags['harness-check']
+      fput 'harness 20' unless Flags['harness-check'].include?('effortlessly')
+      Flags.reset('harness-check')
+    end
+    result = bput('pow', pattern)
+    result =~ pattern
+    if Flags['pattern-shift']
+      Flags.reset('pattern-shift')
+    else
+      move Regexp.last_match[:direction] unless Flags['pattern-shift']
```
Comment (rpherbig) [inline bescort.lic]: Do you need this `unless` check? This line is inside the else case already.

File: bescort.lic [hunk 2]
```diff
@@ -524,6 +530,101 @@ class Bescort
...
+    end
+  end
+
+  def know_shard?(shard_name)
+    result = bput('recall heavens grazhir', shard_name, 'Roundtime')
+    return false if result == 'Roundtime'
+    true
```
Comment (rpherbig) [inline bescort.lic]: You could combine lines 550 and 551 into `result != 'Roundtime'`

File: bescort.lic [hunk 3]
```diff
@@ -524,6 +530,101 @@ class Bescort
...
+    source_shard = DRRoom.room_objs.find{|obj| obj =~ Regexp.union(mode_to_shard.values)}
+                                   .match(Regexp.union(mode_to_shard.values))
+    if source_shard.nil?
+      echo "You are not in a room with a Grazhir shard."
+      exit
+    end
+
+    fput("prep mg")
+    fput("focus #{source_shard}")
+    waitfor 'You feel fully prepared to cast your spell.'
```
Comment (rpherbig) [inline bescort.lic]: I recommend using `waitcastrt?` instead of `waitfor`

**Comment by jandersson** [comment]
Realized that some rooms don't have the shard name or even a shard in room_obj's, so I went with checking room id's instead

File: bescort.lic [hunk 4]
```diff
@@ -599,26 +612,26 @@ class Bescort
...
-    fput("focus #{source_shard}")
-    waitfor 'You feel fully prepared to cast your spell.'
-    3.times{ fput("harness 20") }
-    fput("cast #{source_shard}")
+    source_shard = room_id_to_shard[Room.current.id]
+    prepare?('mg', 5, symbiosis = false, command='prepare')
+    bput("focus #{source_shard}", 'Roundtime')
+    waitcastrt?
+    3.times{ bput("harness 20", 'You tap') }
+    bput("cast #{source_shard}", 'You are pulled through your unstable Moongate')
```
Comment (rpherbig) [inline bescort.lic]: I encourage you to use the `cast` helper in common-arcana (and add the 'pulled through' message to base-spells).

File: bescort.lic [hunk 5]
```diff
@@ -524,6 +531,114 @@ class Bescort
       end
     end
   end
+
+  def power_walk(pattern)
+    if Flags['harness-check']
+      bput('harness 20', 'You tap') unless Flags['harness-check'].include?('effortlessly')
```
Comment (rpherbig) [inline bescort.lic]: There is a `harness_mana` helper in common-arcana that I recommend you use

Comment (jandersson) [inline bescort.lic]: I dont see a harness_mana in common-arcana

Comment (jandersson) [inline bescort.lic]: Ah ha! you just added it

**Comment by rpherbig** [comment]
You may need to restart dependency to get it, I only added it this morning

**Comment by jandersson** [comment]
In action: http://paste.ubuntu.com/23309184/

**Comment by rcuhljr** [comment]
That's sexy.

**Comment by rpherbig** [comment]
Looks good to me

**Comment by rpherbig** [comment]
@jandersson Do you plan to make any more changes or is this ready to be merged?

**Comment by jandersson** [comment]
All set!

**Pull request merged**